kernel/src/log_segment.rs

```diff
@@ -644,4 +657,217 @@ impl LogSegment {
         );
         Ok(())
     }
+
+    /// Checks if a checkpoint schema contains a usable `add.stats_parsed` field.
+    ///
+    /// This validates that:
+    /// 1. The `add.stats_parsed` field exists in the checkpoint schema
+    /// 2. The types in `stats_parsed` are compatible with the stats schema for data skipping
+    ///
+    /// The `stats_schema` parameter contains only the columns referenced in the data skipping
+    /// predicate. This is built from the predicate and passed in by the caller.
+    ///
+    /// Returns `false` if stats_parsed doesn't exist or has incompatible types.
+    fn schema_has_compatible_stats_parsed(
+        checkpoint_schema: &Schema,
+        stats_schema: &StructType,
+    ) -> bool {
+        // Get add.stats_parsed from the checkpoint schema
+        let Some(stats_parsed) = checkpoint_schema
+            .field("add")
+            .and_then(|f| match f.data_type() {
+                DataType::Struct(s) => s.field("stats_parsed"),
+                _ => None,
+            })
+        else {
+            return false;
+        };
+
+        // Get minValues struct to check column types.
+        // We only need to check minValues since minValues and maxValues are always written
```

Comment:
what would happen if writer messed up and this wasn't true (panic or reasonable error?)